src/npc.cpp

```diff
@@ -2045,6 +2045,38 @@ void npc::die( Creature *nkiller )
     place_corpse();
 }
 
+std::string npc_attitude_id( npc_attitude att )
+{
+    static const std::unordered_map<npc_attitude, std::string> npc_attitude_ids = {
```

Comment:
Some compiler versions are unhappy with this, try using a std::map instead.

Reply:
Let's try, but that's kinda strange - it was a direct copy from some other place in the code and should be working.

Maybe I need to add hash function overload for `npc_attitude`?

```CPP
namespace std
{
template <>
struct hash<npc_attitude> {
    std::size_t operator()( const npc_attitude &k ) const {
        return k;
    }
};
} // namespace std
```